### PR TITLE
Completion menu rows - clean up completion menu display

### PR DESCRIPTION
**Pull request by bobhy**
Supercedes #3815, but doesn't include all the fixes yet.  When done, intended to address #3775 and #3384 
Also requires an as-yet-unreleased PR, prompt-toolkit/python-prompt-toolkit#1271.

@anki-code, @ju-kreber, this should address the height-of-the-completion-menu issue as is.  You'll need to install the above version of prompt-toolkit, but can then set $COMPLETION_MENU_ROWS to a reasonable positive number and get no more than that number of completion rows wherever the prompt is in the screen.  There are some other display fixes that should look better, as well..  But no changes yet for anything to do with completions when file name contains spaces, I wanted to get this out of the way first.

And if it's working for you, we can jointly lobby @jonathanslenders  to accept the ptk PR.

### CODE REVIEW

**Comment by scopatz** [comment]
Thanks for putting this in @bobhy!

**Comment by bobhy** [comment]
Fixed backward compatibility: setting $RESERVE_SPACE_FOR_MENU` is used if explicitly set (overrides `$COMPLETION_MENU_ROWS`).  Resulting value is fed to `PromptSession.prompt()` parameter `completion_menu_rows` if it has it, otherwise to parameter `reserve_space_for_menu`.

**Comment by bobhy** [comment]
This PR could now be merged, it will work with the released version of prompt-toolkit (though it will not work any *better* with that version than the master branch does).  It really only adds value if you side-load the unreleased prompt-toolkit/python-prompt-toolkit#1271.  I'm continuing to work on a xonsh-only version of this, but that involves pulling in pieces of PTK into Xonsh so we can tweak its behavior.  Slow going!  Alternatively, we could put the changes in the vended version of PTK, but that would constitute a fork, and I'd recommend against it.

**Comment by scopatz** [comment]
I would love it if some other member of @xore could find some time to review this.

File: xonsh/environ.py [hunk 1]
```diff
@@ -847,10 +847,18 @@ def DEFAULT_VARS():
...
             "prompt-toolkit shell.",
         ),
+        "RESERVE_SPACE_FOR_MENU": Var(
+            is_int,
+            int,
+            str,
+            DefaultNotGiven,
+            "Additional space to reserve at bottom of screen for tab-completions menu. "
+            "Only affects prompt-toolkit shell.  Deprecated.  See ``$COMPLETIONS_MENU_ROWS``.",
+        ),
```
Comment (gforsyth) [inline xonsh/environ.py:861]: I'm deeply confused -- this adds a new env-var that is already deprecated?

Comment (laloch) [reply xonsh/environ.py:861]: @gforsyth, I'm going to stick that quote on my fridge :))

**Comment by ju-kreber** [comment]
With the prompt-toolkit fix, this works for me and successfully fixes issue #3843.

**Comment by jnoortheen** [comment]
maybe late to the show. I was annoyed with this behaviour as well. https://github.com/xonsh/xonsh/pull/4477 is simple but seems to  fix the issue for me. @bobhy Can you test the same?

**Comment by jnoortheen** [comment]
I am closing this to clean-up as the said issues were closed by #4477 .

**Pull request closed without merging**